drivers/ipm/ipm_handlers.c

```diff
@@ -8,13 +8,11 @@
 #include <drivers/ipm.h>
 
 static inline int z_vrfy_ipm_send(const struct device *dev, int wait,
-				  uint32_t id,
-				  const void *data, int size)
+				  struct ipm_msg *msg)
 {
 	Z_OOPS(Z_SYSCALL_DRIVER_IPM(dev, send));
-	Z_OOPS(Z_SYSCALL_MEMORY_READ(data, size));
-	return z_impl_ipm_send((const struct device *)dev, wait, id,
-			       (const void *)data, size);
+	Z_OOPS(Z_SYSCALL_MEMORY_READ(msg->data, msg->size));
```

Comment:
Probably need to check if `msg` itself can be read.

Reply:
done